lib/scenic/theme.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Scenic.Theme do
+  @moduledoc """
+  A Theme is a collection of settings for things like font, color pallettes, and so forth.
+  """
+  defstruct font: nil, pallette: %{}, name: nil
+
+  @type pallette :: %{
+          atom() => Scenic.Color.t()
+        }
+
+  @typedoc """
+  A Theme is a collection of settings for things like font, color pallettes, and so forth.
+
+  The available themeable properties are:
+
+  * `name` is the atom identifying a theme.
+  * `font` is a String of the font that a theme should use, or nil if no preference.
+  * `pallette` is a map mapping atoms like `:primary` or `:secondary` to a `Scenic.Color`.
+  """
+  @type t :: %__MODULE__{
```

Comment:
I'm not sure what Scenic's library policy is, but I'm a big fan of https://hex.pm/packages/typed_struct. If we pulled it in, it would make this type much more convenient to write.

Reply:
I think it'd be cool to add that all in a big typespec pass later!

Reply:
Yeah OK. I hadn't seen typed_struct before. Looks very cool. I'm open to bringing it in.